text/000-cli-redesign.md

```diff
@@ -1 +1,114 @@
+
+## What use cases does it support? 
+Developers who use the Gatsby CLI to start, develop, and build Gatsby sites.
+
+## What is the expected outcome?
+The expected outcome is that the Gatsby CLI will adhere to common sense usability and accessibility standards and will also help people start, develop, and build Gatsby sites faster and with less frustration.
+
+Here are the “user can” statements we want to be true after this redesign:
+Contributors and Ink team members can decide which changes to merge or not through usability heuristics and style principles
+Users can know what an error is and how to solve it
+Users can know the current state of the system
+Users can know when they need to take action and how to take that action
+
+## Data collection outcomes
+See this [RFC on Error surveying & improvement](https://github.com/gatsbyjs/rfcs/pull/37)
+
+## Objectives
+- Increase the rate of adoption - more Gatsby CLI users!
+- Make sure users can reach their goals through the CLI
+- CLI should embody the principles in the CLI style guide
+
+## Anti-goals
+- losing Gatsby users
+- creating frustration or making tasks take more time
+
+# Detailed design
+We’d like to create a [CLI style guide](https://docs.google.com/document/d/1rv7tBUeqHN4t_UecuEAKVCWAVzQoCmcDoONBmf3HggI/edit?usp=sharing) and implement its principles.
+We'd also like to make the changes outlined in the section titled below, called "List of top action items"
+A related RFC is the [Error surveying & improvement RFC](https://github.com/gatsbyjs/rfcs/pull/37)
+
+## List of first hypotheses to test
+- [ ]  If we clear the Gatsby CLI screen to only display success messages or error messages, people will save time in finding the right thing
+- [ ]  If people use a Gatsby CLI quiet mode, people will find and fix errors faster (quiet mode would make errors more discoverable)
+- [ ] If the Gatsby CLI suggests the correct spellings for misspelled Gatsby commands, people will save time and be happier. For example, when people need help, we could make it more easily discoverable so they don’t have to type multiple commands (e.g. `gatsby --help` then `gatsby build  --help`). Possible solution: provide help information for a variety of inputs `gatsby h`, `gatsby`, `gatsby help`, etc. and update our did-you-mean file to dogfood the correct commands
+- [ ] If we prevent same error message over and over again, people will be less confused and fix the error faster
+- [ ] Including createPages page count in output can help users optimize for speed and catch errors faster
+- [ ] When people type the wrong thing in the browser (e.g. “localhost:8000/_ _ graphiql), we could either redirect to "correct" one or just support multiple different cases. This will help them spend less time looking for the right link.
```

Comment:
Is this point really bound to CLI redesign ?

Reply:
probably not, good point. Just a related workflow (e.g. people shift from CLI to browser pretty frequently when running `gatsby develop`. This context switching is actually something that is pretty good to avoid at all costs, from a design perspective